pyccel/codegen/printing/ccode.py

```diff
@@ -412,12 +412,14 @@ def arrayFill(self, expr):
         lhs = expr.lhs
         code_init = ''
         declare_dtype = self.find_in_dtype_registry(self._print(rhs.dtype), rhs.precision)
-
+        dtype = self.find_in_ndarray_type_registry(self._print(rhs.dtype), rhs.precision)
+        dtype = dtype[3:]
```

Comment:
Why do we cut the first 3 characters? Should we have a type_registry instead of a `ndarray_type_registry`?

Reply:
yeah maybe a type registry will be better and we add the `nd_` prefix manually.

Reply:
I think this can be a separate issue/PR

Reply:
https://github.com/pyccel/pyccel-cuda/issues/17